src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -662,6 +668,13 @@ NewTerminalArgs AppCommandlineArgs::_getNewTerminalArgs(AppCommandlineArgs::NewT
         args.AppendCommandLine(_appendCommandLineOption);
     }
 
+    bool inheritEnv = hasCommandline;

```

Comment:
of interest: default to `inherit` if we do have a commandline. If the user passed the arg in manually, then use the arg.